src/modules/jcms_rest/src/Plugin/rest/resource/JobAdvertListRestResource.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Drupal\jcms_rest\Plugin\rest\resource;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\jcms_rest\Response\JCMSRestResponse;
+use Drupal\node\Entity\Node;
+use Drupal\node\NodeInterface;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Provides a resource to get view modes by entity and bundle.
+ *
+ * @RestResource(
+ *   id = "job_advert_list_rest_resource",
+ *   label = @Translation("Job advert list rest resource"),
+ *   uri_paths = {
+ *     "canonical" = "/job-adverts"
+ *   }
+ * )
+ */
+class JobAdvertListRestResource extends AbstractRestResourceBase {
+
+  /**
+   * Responds to GET requests.
+   *
+   * Returns a list of bundles for specified entity.
+   *
+   * @throws \Symfony\Component\HttpKernel\Exception\HttpException
+   *   Throws exception expected.
+   *
+   * @todo - elife - nlisgo - Handle version specific requests
+   */
+  public function get() {
+    $base_query = \Drupal::entityQuery('node')
+      ->condition('status', NodeInterface::PUBLISHED)
+      ->condition('changed', \Drupal::time()->getRequestTime(), '<')
+      ->condition('type', 'job_advert');
+
+    $this->filterShow($base_query);
+    $count_query = clone $base_query;
+    $items_query = clone $base_query;
+    $response_data = [
+      'total' => 0,
+      'items' => [],
+    ];
+    $nodes = [];
+    if ($total = $count_query->count()->execute()) {
+      $response_data['total'] = (int) $total;
+      $this->filterPageAndOrder($items_query);
+      $nids = $items_query->execute();
+      $nodes = Node::loadMultiple($nids);
+      if (!empty($nodes)) {
+        foreach ($nodes as $node) {
+          $response_data['items'][] = $this->getItem($node);
+        }
+      }
+    }
+    $response = new JCMSRestResponse($response_data, Response::HTTP_OK, ['Content-Type' => $this->getContentType()]);
+    $response->addCacheableDependencies($nodes);
```

Comment:
does this mean `invalidate cache if one of these nodes is updated`?

Reply:
That is what is supposed to happen. But we seem to be having issues with that in production. Disabling cache entirely is the priority so I don't want to spend any more time working why the cache isn't behaving as expected.

Reply:
No problem, just looking to understand more